src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -319,32 +319,22 @@
      *
      * @param login the login of the subjectDTO for which to retrieve the revisions
      * @return the ResponseEntity with status 200 (OK) and with body the subjectDTO, or with status
-     *         404 (Not Found)
+     * 404 (Not Found)
      */
     @GetMapping("/subjects/{login:" + Constants.ENTITY_ID_REGEX + "}/revisions")
     @Timed
     public ResponseEntity<List<RevisionDTO>> getSubjectRevisions(
             @ApiParam Pageable pageable, @PathVariable String login) throws NotAuthorizedException {
         log.debug("REST request to get revisions for Subject : {}", login);
-        SubjectDTO subject = subjectService.getLatestRevision(login);
-        Page<RevisionDTO> page = revisionService.getRevisionsForEntity(pageable,
-                subjectMapper.subjectDTOToSubject(subject));
-        SubjectDTO blank = new SubjectDTO();
-        RadarToken token = getJWT(servletRequest);
-        // iterate the revisions, if we don't have read permission for the subject in a given
-        // revision, swap the subject with a 'blank' subject so no information gets leaked
-        page = page.map(rev -> token.hasPermissionOnSubject(SUBJECT_READ,
```

Comment:
@yatharthranjan @blootsvoets @nivemaham Would it be alright to remove this mapping and just check for permissions of the subject once since the revision results are only the specified subject's revisions (in `getRevisionsForEntity`)?

Reply:
It could be that this is due to subjects moving from one project to another. We generally don't do this, so it might be fine to remove it...